storage/filestorage/filestorage.go

```diff
@@ -108,6 +113,7 @@ func (s *FileStorage) Put(_ context.Context, image string, blob *imagor.Blob) (e
 
 // Delete implements imagor.Storage interface
 func (s *FileStorage) Delete(_ context.Context, image string) error {
+	defer prometheus.NewTimer(storage.OperationHistorgram.WithLabelValues("file", "delete")).ObserveDuration()
```

Comment:
[A little copying is better than a little dependency.](https://go-proverbs.github.io/)

Reply:
@nicholascapo you should check out the [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) to understand more about the conventions regarding this library.

Reply:
I guess this is only implied, but I *am* implying that Prometheus is "the right way" to do metrics. :smile:

I don't thinks its a great idea to make these generic or abstract, in such a way that different metrics backend could be swapped out.

I think metrics should be encapsulated by the operational code, so that importing the package causes metrics to be collected.

Additionally if the metrics collection code is here, very close to the actual functionality, it makes it much easier for an engineer to move from an alert or dashboard to the code that is being monitored.

Reply:
Users of New Relic, Datadog, Dynatrace, Splunk, Cloudwatch... and so on, would also claim their chosen metrics library is "the right way" to do metrics....